File: deal-mgmt/app/schemas/negotiation_issue.py
"""협상 이견 스키마 — AI 조항 수정 제안 포함."""

from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field

from app.models.enums import NegotiationIssuePriority, NegotiationIssueStatus


class NegotiationIssueOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    transaction_id: uuid.UUID
    meeting_id: uuid.UUID | None = None
    title: str
    clause_reference: str | None = None
    category: str | None = None
    our_position: str | None = None
    counterpart_position: str | None = None
    legal_review: str | None = None
    ai_suggestion: str | None = None
    ai_suggestion_rationale: str | None = None
    status: NegotiationIssueStatus
    priority: NegotiationIssuePriority
    resolution: str | None = None
    resolved_at: str | None = None
    created_by_email: str | None = None
    created_at: datetime
    updated_at: datetime


class NegotiationIssueCreate(BaseModel):
    meeting_id: uuid.UUID | None = None
    title: str = Field(..., min_length=1, max_length=300)
    clause_reference: str | None = None
    category: str | None = None
    our_position: str | None = None
    counterpart_position: str | None = None
    legal_review: str | None = None
    status: NegotiationIssueStatus = NegotiationIssueStatus.OPEN
    priority: NegotiationIssuePriority = NegotiationIssuePriority.MEDIUM


class NegotiationIssueUpdate(BaseModel):
    meeting_id: uuid.UUID | None = None
    title: str | None = Field(None, min_length=1, max_length=300)
    clause_reference: str | None = None
    category: str | None = None
    our_position: str | None = None
    counterpart_position: str | None = None
    legal_review: str | None = None
    status: NegotiationIssueStatus | None = None
    priority: NegotiationIssuePriority | None = None
    resolution: str | None = None
    resolved_at: str | None = None


class NegotiationIssueListResponse(BaseModel):
    items: list[NegotiationIssueOut]
    total: int


class AIClauseSuggestionResponse(BaseModel):
    suggested_text: str
    rationale: str
    confidence: float

File: amic-platform/src/modules/ma/types/negotiation_issue.ts
// 협상 이견 추적 타입

export type NegotiationIssueStatus = "OPEN" | "IN_PROGRESS" | "AGREED" | "DEFERRED" | "DEADLOCKED";
export type NegotiationIssuePriority = "CRITICAL" | "HIGH" | "MEDIUM" | "LOW";
export type IssueDecisionStatus = "PENDING" | "CONSIDER_ACCEPTING" | "CANNOT_ACCEPT";

export interface NegotiationIssue {
  id: string;
  transaction_id: string;
  meeting_id: string | null;
  contract_id: string | null;
  title: string;
  clause_reference: string | null;
  category: string | null;
  our_position: string | null;
  counterpart_position: string | null;
  legal_review: string | null;
  ai_suggestion: string | null;
  ai_suggestion_rationale: string | null;
  status: NegotiationIssueStatus;
  priority: NegotiationIssuePriority;
  decision_status: IssueDecisionStatus;
  resolution: string | null;
  resolved_at: string | null;
  linked_issue_ids: string[] | null;
  markup_version_number: number | null;
  created_by_email: string | null;
  created_at: string;
  updated_at: string;
}

export interface NegotiationIssueCreate {
  meeting_id?: string;
  contract_id?: string;
  title: string;
  clause_reference?: string;
  category?: string;
  our_position?: string;
  counterpart_position?: string;
  legal_review?: string;
  status?: NegotiationIssueStatus;
  priority?: NegotiationIssuePriority;
  decision_status?: IssueDecisionStatus;
  linked_issue_ids?: string[];
  markup_version_number?: number;
}

export interface NegotiationIssueUpdate {
  meeting_id?: string;
  contract_id?: string;
  title?: string;
  clause_reference?: string;
  category?: string;
  our_position?: string;
  counterpart_position?: string;
  legal_review?: string;
  status?: NegotiationIssueStatus;
  priority?: NegotiationIssuePriority;
  decision_status?: IssueDecisionStatus;
  resolution?: string;
  resolved_at?: string;
  linked_issue_ids?: string[];
  markup_version_number?: number;
}

export interface NegotiationIssueListResponse {
  items: NegotiationIssue[];
  total: number;
}

export interface AIClauseSuggestionResponse {
  suggested_text: string;
  rationale: string;
  confidence: number;
}
